src/modules/review/review.service.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  BadRequestException,
+  ConflictException,
+  ForbiddenException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { Review } from './entities/review.entity';
+import { CreateReviewDto } from './dto/create.review.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Booking } from '../bookings/entities/booking.entity';
+import { Repository } from 'typeorm';
+import { BookingStatus } from '../../shared/constants/booking-status.constant';
+import { Accommodation } from '../accommodations/entities/accommodations.entity';
+import { withBaseResponse } from '../../shared/utils/with-base-response.util';
+import { BaseResponse } from '../../shared/types/base-response.type';
+
+@Injectable()
+export class ReviewService {
+  constructor(
+    @InjectRepository(Booking)
+    private readonly bookingRepository: Repository<Booking>,
+    @InjectRepository(Review)
+    private readonly reviewRepository: Repository<Review>,
+    @InjectRepository(Accommodation)
+    private readonly accommodationRepository: Repository<Accommodation>,
+  ) {}
+
+  async createReview(
+    accountId: string,
+    createReviewDto: CreateReviewDto,
+  ): Promise<BaseResponse<Review>> {
+    const { accommodationId, bookingId, rating, content } = createReviewDto;
+
+    const booking = await this.bookingRepository.findOne({
+      where: { id: bookingId },
+    });
+    if (!booking) {
+      throw new NotFoundException('Booking not found.');
+    }
+
+    if (booking.status !== BookingStatus.Completed) {
+      throw new BadRequestException(
+        'You can only add a review after your booking has been completed.',
+      );
+    }
+
+    if (accountId !== booking.accountId) {
+      throw new ForbiddenException('You can only review your own booking.');
+    }
+
+    const existingReview = await this.reviewRepository.findOne({
+      where: { accountId, bookingId },
+    });
+    if (existingReview) {
+      throw new ConflictException('Review for this booking already exists.');
+    }
+
+    const accommodation = await this.accommodationRepository.findOne({
+      where: { id: accommodationId },
+    });
+    if (!accommodation) {
+      throw new NotFoundException('Accommodation not found.');
+    }
+
+    const review = this.reviewRepository.create({
+      content,
+      rating,
+      accountId,
+      accommodationId,
+      bookingId,
+    });
+
+    return withBaseResponse({
+      status: 201,
```

Comment:
I think it would be better if the service returned the ```data``` part of the ```withBaseResponse``` only. Then we use ```withBaseResponse``` in the controller.